test/test_buffer.cpp

```diff
@@ -23,17 +23,34 @@
 
 namespace sparrow
 {
-    using buffer_test_type = buffer<uint8_t>;
-    using view_test_type = buffer_view<uint8_t>;
+    using buffer_test_type = buffer<int32_t>;
+    using view_test_type = buffer_view<int32_t>;
+
+    using non_trivial_buffer_test_type = buffer<std::string>;
 
     namespace
     {
-        uint8_t* make_test_buffer(std::size_t size, uint8_t start_value = 0)
+        auto make_test_buffer(std::size_t size, int32_t start_value = 0) -> int32_t*
         {
-            uint8_t* res = new uint8_t[size];
+            int32_t* res = new int32_t[size];
             std::iota(res, res + size, start_value);
             return res;
         }
+
+        auto make_test_buffer_non_trivial(std::size_t size) -> std::string*
+        {
+            std::vector<std::string> vec(size);
+            vec.insert(vec.begin(), vec.end(), vec.end());
+            for (std::size_t i = 0; i < size; ++i)
+            {
+                vec[i] = std::to_string(i);
```

Comment:
Not sure it's important but just in case: `to_string` is subject to locale and that also makes it inefficient, which shouldnt impact things here, but just so you know there is `to_chars` series of algorithms that solve both issues. Probably not worth here as this is all tests.

Reply:
Yes, usualy I use Boost to do that but as you said, it's just the unit tests and with std::_to_string the code is short and expressive.